### DISCUSSION TITLE
warning 2022.1

### DISCUSSION THREAD

**Discussion by genus-giuseppe** [indomus/forum#1220] [HUB: Home Assistant]
salve,
ho questo warming, mi servirebbe una mano per capire cosa modificare prima della nuova versione ho cercato di leggere la guida templating ma non ho bene capito qualche anima pia che mi possa dare una mano?

Template warning: 'timestamp_custom' got invalid input '23:40' when rendering template '{{ (states("sensor.time")) == ((states("sensor.alba_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '23:40', however this template will fail to render in Home Assistant core 2022.1

**Comment by snakuzzo** [comment]
Innanzi tutto si scrive "WARNING" e poi ti consiglierei di utilizzare un titolo più esplicativo per il post.
Tornando al warning (che dovresti postare usando il markdown come da regolamento), il problema è che non hai impostato il default value...ed il log te lo dice in maniera esplicita.

Leggi [QUI](https://www.home-assistant.io/docs/configuration/templating/#time)...nello specifico...

![1](https://user-images.githubusercontent.com/7205371/147090559-d236a62e-f913-4e98-9581-8b1ea9561a63.png)

**Reply by genus-giuseppe** [reply]
il messaggio d'errore mi era chiaro non capisco su cosa va modificato se riesci a darmi  una mano ti sarei grato 


##### value_template: "{{ states('sensor.time') == (states.input_datetime.presa_esterna_ora_spegnimento.attributes.timestamp | int | timestamp_custom('%H:%M', False    )) }}

**Reply by snakuzzo** [reply]
Questo che hai appena postato è diverso da quello che è presente nel primo post.
Quale va preso in considerazione ?

Per quest'ultima cosa che hai scritto il template dovrebbe già essere corretto così (ho un template praticamente uguale e non ho alcun problema), ma ti conviene utilizzare la funzione `state_attr` al posto di `states`

```yaml
value_template: "{{ states('sensor.time') == state_attr('input_datetime.presa_esterna_ora_spegnimento','timestamp') | int | timestamp_custom('%H:%M', False ) }}"
```

**Reply by genus-giuseppe** [reply]
scusa ma per errore ti ho riportato il codice errato

###### value_template: '{{ (states("sensor.time")) == ((states("sensor.alba_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}'

**Reply by snakuzzo** [reply]
Che sensore è `sensor.alba_anticipo_posticipo` ?

**Reply by genus-giuseppe** [reply]
mi permette di anticipare o posticipare l'accensione di un switch in base alba tramonto
#####   - platform: template
    sensors:
      alba_anticipo_posticipo:
        friendly_name: "Alba Anticipo Posticipo"
        icon_template: mdi:weather-sunset-up
        value_template: >-
          {% if is_state('input_select.anticipo_posticipo_alba_1', 'posticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_rising) | float + (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% elif is_state('input_select.anticipo_posticipo_alba_1', 'anticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_rising) | float - (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% endif %}

**Reply by snakuzzo** [reply]
Prova a definire il sensore così...

```yaml
sensor:

- platform: template
  sensors:
    alba_anticipo_posticipo:
      friendly_name: "Alba Anticipo Posticipo"
      icon_template: mdi:weather-sunset-up
      value_template: >-
        {% if is_state('input_select.anticipo_posticipo_alba_1', 'posticipo') %}
          {{ ((as_timestamp(states.sun.sun.attributes.next_rising), 0) | float + (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
        {% elif is_state('input_select.anticipo_posticipo_alba_1', 'anticipo') %}
          {{ ((as_timestamp(states.sun.sun.attributes.next_rising), 0) | float - (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
        {% endif %}
```

**Reply by genus-giuseppe** [reply]
Da sempre lo stesso errore:  

##### Logger: homeassistant.helpers.template
Source: helpers/template.py:1291
First occurred: 20:03:00 (3 occurrences)
Last logged: 20:03:10

Template warning: 'timestamp_custom' got invalid input '17:30' when rendering template '{{ (states("sensor.time")) == ((states("sensor.alba_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '17:30', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'timestamp_custom' got invalid input '17:00' when rendering template '{{ (states("sensor.time")) == ((states("sensor.tramonto_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '17:00', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'timestamp_custom' got invalid input '16:59' when rendering template '{{ (states("sensor.time")) == ((states("sensor.tramonto_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '16:59', however this template will fail to render in Home Assistant core 2022.1

sotto la configurazione dei 2 sensori:

1) questo e' quello modificato
#####   - platform: template
    sensors:
      alba_anticipo_posticipo:
        friendly_name: "Alba Anticipo Posticipo"
        icon_template: mdi:weather-sunset-up
        value_template: >-
          {% if is_state('input_select.anticipo_posticipo_alba_1', 'posticipo') %}
            {{ ((as_timestamp(states.sun.sun.attributes.next_rising), 0) | float + (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% elif is_state('input_select.anticipo_posticipo_alba_1', 'anticipo') %}
            {{ ((as_timestamp(states.sun.sun.attributes.next_rising), 0) | float - (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% endif %}

2) questo rimasto invariato
#####   - platform: template
    sensors:
     tramonto_anticipo_posticipo:
        friendly_name: "Tramonto Anticipo Posticipo"
        icon_template: mdi:weather-sunset-down
        value_template: >-
          {% if is_state('input_select.anticipo_posticipo_tramonto_1', 'posticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_setting) | float + (states.input_datetime.delay_tramonto.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% elif is_state('input_select.anticipo_posticipo_tramonto_1', 'anticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_setting) | float - (states.input_datetime.delay_tramonto.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
            {% endif %}

**Reply by snakuzzo** [reply]
Mi fai vedere anche come hai definito le altre entità utilizzate nel template sensor? Mi riferisco agli `input-select`... così riproduco in test appena posso

**Reply by genus-giuseppe** [reply]
se vuoi ti giro il file completo ancora da sistemare 
##### input_select:
  anticipo_posticipo_alba_1:
    options:
      - "posticipo"
      - "anticipo"
  anticipo_posticipo_tramonto_1:
    options:
      - "posticipo"
      - "anticipo"

**Reply by snakuzzo** [reply]
sì passa tutto il package...forse è meglio

**Comment by genus-giuseppe** [comment]
Grazie per la disponibilita' ho utilizzato e modificato 2 package diversi ed inserendo due sensori alba tramonto, ancora il file e da pulire per mancanza di tempo non ho tolto quello che non  #serviva
[Nuovo documento di testo.txt](https://github.com/indomus/forum/files/7771179/Nuovo.documento.di.testo.txt)

**Reply by genus-giuseppe** [reply]
buongiorno ed auguri di buon Natale
ho quasi risolto aggiungendo ai sensori (default=0)

ma ancora ho questo errore 
#Template warning: 'timestamp_custom' got invalid input '17:03' when rendering template '{{ (states("sensor.time")) == ((states("sensor.tramonto_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '17:03', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'timestamp_custom' got invalid input '23:41' when rendering template '{{ (states("sensor.time")) == ((states("sensor.alba_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}' but no default was specified. Currently 'timestamp_custom' will return '23:41', however this template will fail to render in Home Assistant core 2022.1

**Comment by genus-giuseppe** [comment]
mi rimane solo 1 avviso da sistemare si tratta dell'automazione 

#####   - alias: Spegni luce all'alba con delay             
    initial_state: 'on'
    trigger:
      - platform: template
      
        value_template: '{{ (states("sensor.time")) == ((states("sensor.alba_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}'  



se puoi essere d'aiuto questo ho risolto inserendo (default=0)

######   - platform: template
    sensors:
      alba_anticipo_posticipo:
        friendly_name: "Alba Anticipo Posticipo"
        icon_template: mdi:weather-sunset-up
        value_template: >-
          {% if is_state('input_select.anticipo_posticipo_alba_1', 'posticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_rising) | float(default=0) + (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% elif is_state('input_select.anticipo_posticipo_alba_1', 'anticipo') %}
            {{ (as_timestamp(states.sun.sun.attributes.next_rising) | float(default=0) - (states.input_datetime.delay_alba.attributes.timestamp)) | timestamp_custom("%H:%M") | replace(" 0", "") }}
          {% endif %}

**Reply by snakuzzo** [reply]
Sul default, ti ringrazio ma non mi serve aiuto :)
Detto questo, non ho capito se hai risolto tutto oppure no. In questi giorni non ho possibilità di aiutarti

**Reply by genus-giuseppe** [reply]
mi e' rimasto solo questo fuori 

##### - alias: Spegni luce all'alba con delay
initial_state: 'on'
trigger:
  - platform: template
  
    value_template: '{{ (states("sensor.time")) == ((states("sensor.tramonto_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}'

**Reply by snakuzzo** [reply]
che cosa vorrebbe dire ? non ho capito.
Ribadisco che devi usare il markdown per il codice...te lo sto dicendo dall'inizio.
Non penso ci voglia tutto questo studio per capire come utilizzarlo...è scritto pure nel regolamento del forum -> https://github.com/indomus/forum/blob/main/README.md

**Reply by genus-giuseppe** [reply]
scusami di nuovo sotto il codice  l'automazione cosi funziona ma nel log ho sempre il messaggio d'errore

```
  - alias: Accendi luce al tramonto con delay     
    initial_state: 'on'
    trigger:
      - platform: template
        value_template: '{{ (states("sensor.time")) == ((states("sensor.tramonto_anticipo_posticipo")) | timestamp_custom("%H:%M", False)) }}'

    condition:
```

**Reply by genus-giuseppe** [reply]
sto effettuando varie prove ma ancora non ne vengo a capo.
